### PR TITLE
Fix storage bugs and make storage integration tests run

### LINKED ISSUE #2056: When attempting to fetch file metadata, the file itself is returned not the metadata

### PR DESCRIPTION
**Pull request by imsdu**
Fixes #2056 #2061 #2062 #2077

### CODE REVIEW

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageLinkFile.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+class S3StorageLinkFile(storage: S3Storage)(implicit as: ActorSystem) extends LinkFile {
+
+  import as.dispatcher
+
+  private val fileNotFoundException = new IllegalArgumentException("File not found")
+
+  override def apply(key: Uri.Path, description: FileDescription): IO[StorageFileRejection, FileAttributes] = {
+    val attributes    = S3Attributes.settings(storage.value.toAlpakkaSettings)
+    val location: Uri = s"${storage.value.address(storage.value.bucket)}/$key"
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageLinkFile.scala:27]: why is used the string conversion here and not the actual Uri conversion?
```scala
import ch.epfl.bluebrain.nexus.delta.rdf.syntax.sdk.syntax._

val location: Uri = storage.value.address(storage.value.bucket) / key
```

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala [hunk 2]
```diff
@@ -259,7 +259,7 @@ object StoragesRoutes {
       case IncorrectRev(_, _)                => StatusCodes.Conflict
       case WrappedProjectRejection(rej)      => rej.status
       case WrappedOrganizationRejection(rej) => rej.status
-      case StorageNotAccessible(_, _)        => StatusCodes.Forbidden
+      case StorageNotAccessible(_, _)        => StatusCodes.BadRequest
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala:262]: Why? If it is not accessible it is as if the client wouldn't have permissions

**Review by umbreak — CHANGES_REQUESTED** [review]

**Review by bogdanromanx — DISMISSED** [review]

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageLinkFile.scala (hunk 1 above)
Comment (imsdu) [reply delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageLinkFile.scala:27]: I forgot about it

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala (hunk 2 above)
Comment (imsdu) [reply delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala:262]: `StorageNotAccessible` is raised when volumes / bucket are invalid no ? It is not another rejection when there are no permissions ?

Comment (umbreak) [reply delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala:262]: Now it seems to me more like a NotFound.
But yeah, bad request is fine I guess

**Review by umbreak — APPROVED** [review]

**Pull request merged**